lib/msf/core/modules/loader/directory.rb

```diff
@@ -32,13 +32,11 @@ def loadable?(path)
   def each_module_reference_name(path, opts={})
     whitelist = opts[:whitelist] || []
     ::Dir.foreach(path) do |entry|
+
       full_entry_path = ::File.join(path, entry)
       type = entry.singularize
 
-      unless ::File.directory?(full_entry_path) and
-             module_manager.type_enabled? type
-        next
-      end
+      next if !::File.directory?(full_entry_path) || !module_manager.type_enabled?(type)
```

Comment:
Merge conflicted here, and my hackishness was no better IMO, but figure we might make it more readable as 
```
next unless ::File.directory?(full_entry_path) && module_manager.type_enabled?(type)
```